javalib/src/main/scala/java/time/Year.scala

```diff
@@ -0,0 +1,203 @@
+        val era = getLong(ERA)
+        if (era != value) Year.of(1 - year)
+        else this
+
+      case _: ChronoField =>
+        throw new UnsupportedTemporalTypeException(s"Unsupported field: $field")
+
+      case _ =>
+        field.adjustInto(this, value)
+    }
+  }
+
+  def plus(amount: Long, unit: TemporalUnit): Year = unit match {
+    case YEARS     => plusYears(amount)
+    case DECADES   => plusYears(MathJDK8Bridge.multiplyExact(amount, 10))
+    case CENTURIES => plusYears(MathJDK8Bridge.multiplyExact(amount, 100))
+    case MILLENNIA => plusYears(MathJDK8Bridge.multiplyExact(amount, 1000))
+    case ERAS      =>
+      val era = getLong(ERA)
+      `with`(ERA, MathJDK8Bridge.addExact(era, amount))
+
+    case _: ChronoUnit =>
+      throw new UnsupportedTemporalTypeException(s"Unsupported unit: $unit")
+
+    case _ =>
+      unit.addTo(this, amount)
+  }
+
+  def plusYears(amount: Long): Year = {
+    if (amount == 0) {
+      this
+    } else {
+      val newYear = MathJDK8Bridge.addExact(year, amount)
+      Year.of(YEAR.checkValidIntValue(newYear))
+    }
+  }
+
+  override def minus(amount: Long, unit: TemporalUnit): Year = {
+    if (amount != Long.MinValue) plus(-amount, unit)
+    else plus(Long.MaxValue, unit).plus(1, unit)
+  }
+
+  def minusYears(amount: Long): Year = minus(amount, YEARS)
+
+  // Not implemented
+  // def query[R](query: TemporalQuery[R]): R
+
+  def adjustInto(temporal: Temporal): Temporal =
+    temporal.`with`(YEAR, year)
+
+  def until(end: Temporal, unit: TemporalUnit): Long = {
+    val other = Year.from(end)
+    val yearsDiff: Long = other.getValue.toLong - year
+
+    unit match {
+      case YEARS     => yearsDiff
+      case DECADES   => yearsDiff / 10
+      case CENTURIES => yearsDiff / 100
+      case MILLENNIA => yearsDiff / 1000
+      case ERAS      => other.getLong(ERA) - getLong(ERA)
+
+      case _: ChronoUnit =>
+        throw new UnsupportedTemporalTypeException(s"Unsupported unit: $unit")
+
+      case _ => unit.between(this, other)
+    }
+  }
+
+  def atDay(dayOfYear: Int): LocalDate =
+    LocalDate.ofYearDay(year, dayOfYear)
+
+  // TODO
+  // def atMonth(month: Month): YearMonth = YearMonth.of(year, month)
+
+  // TODO
+  // def atMonth(month: Int): YearMonth = YearMonth.of(year, month)
+
+  def atMonthDay(monthDay: MonthDay): LocalDate = monthDay.atYear(year)
+
+  def compareTo(other: Year): Int = year - other.getValue
+
+  def isAfter(other: Year): Boolean = compareTo(other) > 0
+
+  def isBefore(other: Year): Boolean = compareTo(other) < 0
+
+  override def equals(other: Any): Boolean = other match {
+    case that: Year => year == that.getValue
+    case _ => false
+  }
+
+  override def hashCode(): Int = year.hashCode()
+
+  override def toString: String = s"$year"
+
+  // Not implemented
+  // def format(formatter: DateTimeFormatter): String
+
+}
+
+object Year {
+  import ChronoField._
+
+  val MIN_VALUE: Int = -999999999
+  val MAX_VALUE: Int = 999999999
```

Comment:
Now that we have these values you should rewrite the definition of `ChronoField.YEAR` to use them. 
```scala
  final val YEAR = new ChronoField("Year", 26,
      ValueRange.of(Year.MIN_VALUE, Year.MAX_VALUE), YEARS, FOREVER, isDateBasedFlag)
```

This is to ensure coherence with `YEAR.checkValidIntValue(year)` bellow (l.190).